tests/test_trusted-execution.yml

```
@@ -0,0 +1,132 @@
+---
+- name: Basic sanity test for fapolicyd
+  hosts: localhost
+  vars:
+    binaries_directory: "/var/tmp/executable_binaries"
+    source_directory: "{{ binaries_directory }}/source"
+    source_file: "{{ source_directory }}/main.c"
+    exe1: "{{ binaries_directory }}/exe1"
+    exe2: "{{ binaries_directory }}/exe2"
+    user: "testuser"
+
+  tasks:
+    - name: Create {{ binaries_directory }} directory
+      file:
+        path: "{{ binaries_directory }}"
+        state: directory
+        mode: '0755'
+
+    - name: Create {{ source_directory }} directory
+      file:
+        path: "{{ source_directory }}"
+        state: directory
+        mode: '0755'
+
+    - name: Create C source code (binary1)
+      copy:
+        content: |
+          int main(void) { 
+              return 0; 
+          }
+        dest: "{{ source_file }}"
+
+    - name: Install build essentials (for RHEL-based systems)
+      block:
+        - name: Install GCC
+          yum:
+            name: gcc
+            state: present
+
+        - name: Install glibc-devel (for standard library headers)
+          yum:
+            name: glibc-devel
+            state: present
+    
+    - name: Compile C programs
+      command: "gcc -o {{ exe1 }} {{ source_file }}"
+      ignore_errors: true
+
+    - name: Compile C programs (exe2)
+      command: "gcc -g -o {{ exe2 }} {{ source_file }}"
+      ignore_errors: true
+
+    - name: Create a new user
+      user:
+        name: "{{ user }}"
+        state: present
+        shell: /bin/bash
+
+    - name: Run the role
+      include_role:
+        name: linux-system-roles.fapolicyd
+      vars:
+        fapolicyd_setup_enable_service: true
+        fapolicyd_setup_integrity: sha256
+        fapolicyd_setup_trust: rpmdb,file
+        fapolicyd_add_trusted_file:
+          - /etc/passwd
+          - /etc/fapolicyd/fapolicyd.conf
+          - /etc/krb5.conf
+          - "{{ exe1 }}"
+
+    - name: Run trusted binary "{{ exe1 }}"
+      shell: su - {{ user }} -c "{{ exe1 }}"
+
+    - name: Run untrusted binary "{{ exe2 }}"
+      shell: su - {{ user }} -c "{{ exe2 }}"
+      register: exe2_result
+      ignore_errors: true
+      failed_when: "exe2_result.rc != 126"
+    
+    - name: Replace binary "{{ exe1 }}" with "{{ exe2 }}"
+      shell: cat "{{ exe2 }}" > "{{ exe1 }}"
```

```suggestion
      shell: cat {{ exe2 | quote }} > "{{ exe1 | quote }}"
```

If quotation is needed.